be/src/common/format_ip.h

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <cstdint>
+#include <cstring>
+
+constexpr size_t IPV4_BINARY_LENGTH = 4;
+constexpr int IPV4_MAX_OCTET_VALUE = 255; //max value of octet
+constexpr size_t IPV4_OCTET_BITS = 8;
+constexpr size_t DECIMAL_BASE = 10;
+
+namespace starrocks {
+
+inline bool is_numeric_ascii(char c) {
+    // This is faster than
+    // return UInt8(UInt8(c) - UInt8('0')) < UInt8(10);
+    // on Intel CPUs when compiled by gcc 8.
+    return (c >= '0' && c <= '9');
+}
+
+inline bool parse_ipv4(const char* pos, int64_t& dst, int64_t first_octet = -1) {
+    if (pos == nullptr || first_octet > IPV4_MAX_OCTET_VALUE) {
+        return false;
+    }
+
+    int64_t result = 0;
+    int offset = (IPV4_BINARY_LENGTH - 1) * IPV4_OCTET_BITS;
+
+    if (first_octet >= 0) {
+        result |= first_octet << offset;
+        offset -= IPV4_OCTET_BITS;
+    }
+
+    for (; true; offset -= IPV4_OCTET_BITS, ++pos) {
```

Comment:
We need to check pos length.

Reply:
...
size_t length = std::strlen(pos);
if (length < 7 || length > 15) { 
     return false;
}

for (; true; offset -= IPV4_OCTET_BITS, ++pos) {
...
@stdpain Do you think this approach effectively addresses the concern you mentioned?I'm open to any feedback or additional insights you might have.Thank you!

Reply:
the length should be str_value.get_size()
```
        auto str_value = str_viewer.value(row);
        int64_t parsed_result;
        if (try_parse_ipv4(str_value.get_data(), str_value.get_size(), parsed_result)) {
```

Reply:
@stdpain I have modified the code according to your suggestions and tested it locally, confirming that everything runs smoothly. Could you please review the PR again? Thank you!